test/integration-test/src/tests/xdp.rs

```diff
@@ -1,15 +1,86 @@
-use std::{net::UdpSocket, time::Duration};
+use std::{
+    ffi::CStr, mem::MaybeUninit, net::UdpSocket, num::NonZeroU32, ptr::NonNull, time::Duration,
+};
 
 use aya::{
-    maps::{Array, CpuMap},
+    maps::{Array, CpuMap, XskMap},
     programs::{Xdp, XdpFlags},
     Bpf,
 };
 use object::{Object, ObjectSection, ObjectSymbol, SymbolSection};
 use test_log::test;
+use xdpilone::{BufIdx, IfInfo, Socket, SocketConfig, Umem, UmemConfig};
 
 use crate::utils::NetNsGuard;
 
+#[test]
+fn af_xdp() {
+    let _netns = NetNsGuard::new();
+
+    let mut bpf = Bpf::load(crate::REDIRECT).unwrap();
+    let mut socks: XskMap<_> = bpf.take_map("SOCKS").unwrap().try_into().unwrap();
+
+    let xdp: &mut Xdp = bpf
+        .program_mut("redirect_sock")
+        .unwrap()
+        .try_into()
+        .unwrap();
+    xdp.load().unwrap();
+    xdp.attach("lo", XdpFlags::default()).unwrap();
+
+    #[repr(align(4096))]
+    struct PacketMap(MaybeUninit<[u8; 4096]>);
+
+    let mem = Box::new(PacketMap(MaybeUninit::uninit()));
```

Comment:
nit: you're never freeing this

Reply:
@alessandrod are there any chance the tests will ever run on apple silicon? pages on M1/M2 are 16k instead of 4k, so it may be different for the `PacketMap` newtype. Is it worth it to change the `repr` of the type?

Reply:
@alessandrod I align'ed to 16384 and I'm calling it a day.